File: src/1-Problem09.pas
program Problem9;

var
    SS1,SS2 : char; (* satuan suhu termometer*)
    suhu : real;
    result : real;
    valid : boolean;
begin
    valid := True;
    write('Masukan satuan suhu pada termometer: '); readln(SS1);
    write('Masukan besar suhu : '); readln(suhu);
    write('Masukan satuan yang ingin diketahui: '); readln(SS2);
    Case SS1 of
        'C' :
          begin
            if ( SS2 = 'R' ) then
              begin
                result := 4 / 5 * suhu;
              end
            else if ( SS2 = 'F' ) then
              begin
                result := ( 9 / 5 * suhu ) + 32;
              end
            else
              begin
                writeln('Masukan tidak valid'); 
                valid := False;
              end;
          end;
        'R' : 
          begin
            if ( SS2 = 'C' ) then
              begin
                result := 5 / 4 * suhu;
              end
            else if ( SS2 = 'F' ) then
              begin
                result := ( 9 / 4 * suhu ) + 32;
              end
            else
              begin
                writeln('Masukan tidak valid'); 
                valid := False;
              end;
          end;
        'F' :
          begin
            suhu := suhu - 32;
            if ( SS2 = 'C' ) then
              begin
                result := 5 / 9  * suhu;
              end
            else if ( SS2 = 'R' ) then
              begin
                result := 4 / 9 * suhu;
              end
            else
              begin
                writeln('Masukan tidak valid'); 
                valid := False;
              end;
          end;
    else
        writeln('Masukan tidak valid');
        valid := False;
    end;
    if ( valid ) then
        writeln('Besar suhu dalam ',SS2,' : ', result:0:2);
end.

File: src/1-Problem09.cpp
#include <iostream>
using namespace std;

int main()
{
	char in, out;
	double awal, akhir;
	bool valid = true;

	/* Menerima jenis suhu masukan */
	cout << "Masukan satuan suhu termometer: ";
	cin >> in;

	/* Menerima besar suhu masukan */
	cout << "Masukan besar suhu: ";
	cin >> awal;
	
	/* Menerima jenis suhu keluaran */
	cout << "Masukan satuan yang ingin diketahui: ";
	cin >> out;
	
	/* Kalkulasi */
	if (in == out) {
		/* tidak ada perubahan */
	}
	else {
		if (in == 'C') {
			switch (out) {
				case 'R':
					akhir = awal * 4 / 5;
					break;
				case 'F':
					akhir = (awal * 9 / 5) + 32;
					break;
				default :
					valid = false; 
			}
		}
		else if (in == 'R') {
			switch (out) {
				case 'C':
					akhir = awal * 5 / 4;
					break;
				case 'F':
					akhir = (awal * 9 / 4) + 32;
					break;
				default :
					valid = false; 
			}
		}
		else if (in == 'F') {
			switch (out) {
				case 'R':
					akhir = (awal - 32) * 4 / 9;
					break;
				case 'C':
					akhir = (awal - 32) * 5 / 9;
					break;
				default :
					valid = false; 
			}
		}
		else {
			// Format Input Salah
			valid = false;
		}
	}
	
	/* Menampilkan hasil */
	cout << "Besar suhu dalam " << out << ": ";
	if (valid) {
		cout << akhir << endl;
	}
	else {
		cout << "UNKNOWN" << endl;
	}
	return 0;
}